### ISSUE TITLE
UX Suggestion: warn users when commands will take more than 10 seconds or so

### ISSUE THREAD

**Issue by nathanleclaire** [docker/machine#230]
Generally I think that people expect commands to run almost instantaneously so I think it'd be good to warn them in commands like `create` that "This might take a few minutes...".

**Comment by ehazlett** [comment]
+1

**Comment by ehazlett** [comment]
+1 to @ahmetalpbalkan

**Comment by janeczku** [comment]
![activityindicator](http://g.recordit.co/XxdLPtDBEc.gif)

Unfinished work. I think it would be best to integrate this in the logger. For that the logger should be a global instance so we can keep track of the state of the progress/activity indicator (e.g. clear the indicator when a new log event occurs and on os.exit).

**Comment by ahmetalpbalkan** [comment]
@janeczku :clap:

**Comment by nathanleclaire** [comment]
@ahmetalpbalkan Will this work compatibly with Windows shells?  `cmd.exe`, msysgit, Powershell?  The Docker client, for instance, currently has a lot of issues with the "real-time" output from `docker pull` etc. in my experience.  I think I might prefer the boot2docker-cli way of just outputting dots.  Still big +1 on telling the user it might take a minute though.

**Comment by ahmetalpbalkan** [comment]
@nathanleclaire if we integrate `docker/pkg/term` correctly to the machine it will work on windows shells as well. We're currently in process of rewriting the ANSI emulation package (and thus the windows parts in `docker/pkg/term`) you can see here: https://github.com/docker/docker/pull/13224.

Once that's done I can help you guys integrate fancy ANSI animations into windows consoles.

Otherwise as is this (and `machine ssh`) won't work on windows right away today.

**Comment by janeczku** [comment]
@nathanleclaire that's because docker's progress indicator is implemented by using ansi escape sequences, which aren't really supported by win shells. The code i am testing should work on any terminal. E.g. msysgit on Windows XP:
http://![msysgit](http://g.recordit.co/tJ7KlbBEsS.gif)

**Comment by ahmetalpbalkan** [comment]
@janeczku true. The ansi emulator we're writing basically will allow stdout/stderr streams that you can set as os.Stdout/os.Stderr and it will capture ANSI sequences and make win32 API calls to achieve the same visual behavior. Please go ahead with your change and once we're done developing the ANSI emulator and integrate it to docker/docker,we're gonna integrate it to machine as well and make windows consoles an officially supported target as well.

**Comment by janeczku** [comment]
@ahmetalpbalkan :+1:
